src/common.c

```diff
@@ -130,6 +130,36 @@ auto_free_char(char** str)
     free(*str);
 }
 
+/**
+ * Closes the file descriptor.
+ *
+ * @param fd file descriptor handle. If NULL, no action is taken.
+ */
+void
+auto_close_gfd(gint* fd)
```

Comment:
Just out of curiosity: why did you decide to have our own version instead of using `g_autofd`?

Reply:
Never knew this existed. I only now found https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3007 when googling. devhelp doesnt seem to contain it?

Reply:
Should I use that or we want the log_error()

Reply:
Phew, good question ... I've never seen a failed `[f]close()`, so I'm not sure if there's any value in it. But maybe that's also only the case because nobody usually checks their return value?
Let's just keep using ours.